encodings/alp/src/alp_rd/compute/take.rs

```diff
@@ -1,18 +1,31 @@
-use vortex_array::compute::{take, TakeFn};
+use vortex_array::compute::{fill_null, take, TakeFn};
 use vortex_array::{Array, IntoArray};
 use vortex_error::VortexResult;
+use vortex_scalar::{Scalar, ScalarValue};
 
 use crate::{ALPRDArray, ALPRDEncoding};
 
 impl TakeFn<ALPRDArray> for ALPRDEncoding {
     fn take(&self, array: &ALPRDArray, indices: &Array) -> VortexResult<Array> {
+        let taken_left_parts = take(array.left_parts(), indices)?;
         let left_parts_exceptions = array
             .left_parts_patches()
             .map(|patches| patches.take(indices))
             .transpose()?
-            .flatten();
+            .flatten()
+            .map(|p| {
+                let values_dtype = p
+                    .values()
+                    .dtype()
+                    .with_nullability(taken_left_parts.dtype().nullability());
+                p.cast_values(&values_dtype)
+            })
+            .transpose()?;
 
-        let taken_left_parts = take(array.left_parts(), indices)?;
+        let right_parts = fill_null(
+            take(array.right_parts(), indices)?,
+            Scalar::new(array.right_parts().dtype().clone(), ScalarValue::from(0)),
+        )?;
```

Comment:
Should add a thing to alp-rd to not require right parts to be nonnullable and just fill in the value with 0 (left parts will be null anyway)